packages/core/contracts/oracle/implementation/DesignatedVotingV2.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../common/implementation/MultiRole.sol";
+import "../../common/implementation/Stakeable.sol";
+import "../interfaces/VotingAncillaryInterface.sol";
+import "../interfaces/FinderInterface.sol";
+import "./Constants.sol";
+
+/**
+ * @title Proxy to allow voting from another address.
+ * @dev Allows a UMA token holder to designate another address to vote on their behalf.
+ * Each voter must deploy their own instance of this contract.
+ */
+contract DesignatedVotingV2 is Stakeable {
+    /****************************************
+     *    INTERNAL VARIABLES AND STORAGE    *
+     ****************************************/
+
+    enum Roles {
```

Comment:
I thought we were trying to move off of multirole, now that OZ has a comparable library?

Note: something that the current version has that would be really desirable to keep in a new version would be for a layman to be able to easily query the owner and designated voter via etherscan. It's a feature we use all the time.

Reply:
I kept it like this as I wanted this to be a minimum viable drop-in for the current system by keeping _all_ logic the same other than adding support for the new staking logic. as a separate PR we can strip out this and move to the OZ stuff but then it should be done in the voting.sol contract at the same.